messenger.go

```diff
@@ -435,6 +440,45 @@ func (m *Messenger) Attachment(to Recipient, dataType AttachmentType, url string
 	return response.Attachment(dataType, url, messagingType, tags...)
 }
 
+// EnableChatExtension set the homepage url required for a chat extension.
+func (m *Messenger) EnableChatExtension(homeURL HomeURL) error {
+	wrap := map[string]interface{}{
+		"home_url": homeURL,
+	}
+	data, err := json.Marshal(wrap)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("POST", MessengerProfileURL, bytes.NewBuffer(data))
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.URL.RawQuery = "access_token=" + m.token
+
+	client := &http.Client{}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	qr := QueryResponse{}
+	err = json.NewDecoder(resp.Body).Decode(&qr)
+	if err != nil {
+		return err
+	}
+
+	if qr.Error != nil {
```

Comment:
I didn't notice before but you should be able to use `checkFacebookError`, see:

https://github.com/paked/messenger/blob/3dda3ecdde73528d57e4f9023312c74c031c5df3/response.go#L71-L82

Also, I see that other methods for settings are defined a bit earlier in this file, you can for example `GreetingSetting` is defined at lines 189-220. Given that, do you think it would make sense to move `EnableChatExtension` closer to these other methods?

Reply:
updated, rebased, squashed :)